ext/vscode/src/services/AzureDevShowProvider.ts

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { IActionContext } from '@microsoft/vscode-azext-utils';
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { createAzureDevCli } from '../utils/azureDevCli';
+import { execAsync } from '../utils/process';
+import { withTimeout } from '../utils/withTimeout';
+
+interface AzureDevService {
+    readonly project: {
+        readonly path: string;
+        readonly language: string;
+    };
+    readonly target: {
+        readonly resourceIds: string[];
+    };
+}
+
+interface AzureDevResource {
+    readonly id: string;
+    readonly type: string;
+}
+
+export interface AzDevShowResults {
+    readonly name: string;
+    readonly services: { readonly [name: string]: AzureDevService };
+    readonly environmentName?: string; // Available in version 0.6.0+
+    readonly resources?: AzureDevResource[];
+}
+
+export interface AzureDevShowProvider {
+    getShowResults(context: IActionContext, configurationFile: vscode.Uri, environmentName?: string): Promise<AzDevShowResults>;
+}
+
+export class WorkspaceAzureDevShowProvider implements AzureDevShowProvider {
+    public async getShowResults(context: IActionContext, configurationFile: vscode.Uri, environmentName?: string): Promise<AzDevShowResults> {
+        const azureCli = await createAzureDevCli(context);
+
+        const configurationFileDirectory = path.dirname(configurationFile.fsPath);
+
+        const command = azureCli.commandBuilder
+            .withArg('show')

```

Comment:
Nit: I wish this command was named `describe app` or something along these lines. `show` begs the "show what?" question.

Reply:
I think since `azd show` isn't really meant to be user-facing, it's OK

Reply:
That's fair